smart-contracts/cosmwasm-contract/src/msg.rs

```diff
@@ -0,0 +1,48 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub supra_pull_contract: String,
+}
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    UpdateSupraContract { supra_pull_contract: String },
+    VerifyOracleProof { proof_bytes: Vec<u8> },
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    /// GetCount returns the current count as a json-encoded number
+    #[returns(CountResponse)]
+    GetSupraPullContract {},
+    // Get price data from pair_id within own storage
+    #[returns(PriceData)]
+    GetPairDataInternal { pair_id: u32 },
+    #[returns(OracleHolder)]
+    GetPairDataSupra { pair_id: u32 },
+}
+
+// We define a custom struct for each query response
+#[cw_serde]
+pub struct CountResponse {
```

Comment:
The struct name does not accurately reflect its purpose or the data it is responding with.